docs: HOWTO run cilium-cli e2e connectivity tests (#25217)
cilium-team/cilium: Documentation/contributing/testing/e2e.rst

.. _testsuite:

End-To-End Connectivity Testing
===============================

Introduction
~~~~~~~~~~~~

Cilium uses `cilium-cli connectivity tests
<https://github.com/cilium/cilium-cli/#connectivity-check>`_
for implementing and running end-to-end tests which test Cilium all the way

> from the API level (e.g. importing policies, CLI) to the datapath (i.e, whether

Reviewer: No Latin abbreviations
```suggestion
from the API level (for example, importing policies, CLI) to the datapath (in order words, whether
```

policy that is imported is enforced accordingly in the datapath).

Running End-To-End Connectivity Tests
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The connectivity tests are implemented in such a way that they can be run against
any K8s cluster running Cilium. The built-in feature detection allows the testing 
framework to automatically skip tests when a required test condition cannot be met

> (e.g., skip the Egress Gateway tests if the Egress Gateway feature is disabled).

Reviewer: No Latin abbreviations
```suggestion
(for example, skip the Egress Gateway tests if the Egress Gateway feature is disabled).
```

Running tests locally
^^^^^^^^^^^^^^^^^^^^^

> Before running the connectivity tests you need to install `Cilium CLI <https://github.com/cilium/cilium-cli#installation>`_.

Reviewer: clarity
```suggestion
Before running the connectivity tests, you must install `Cilium CLI <https://github.com/cilium/cilium-cli#installation>`_.
```

> Alternatively, ``Cilium CLI`` can be manually built and installed by fetching
> ``https://github.com/cilium/cilium-cli``, and then running ``make install``.

Reviewer: Use active voice, comma nit, clarity
```suggestion
Alternatively, you can manually build and install ``Cilium CLI`` by fetching
``https://github.com/cilium/cilium-cli`` and running ``make install``.
```

Next, you need a Kubernetes cluster to run Cilium. The easiest way to create one
is to use `kind <https://github.com/kubernetes-sigs/kind>`_. Cilium provides
a wrapper script which simplifies creating K8s cluster with ``kind``. For example,
to create a cluster consisting of 1 control-plane node, 3 worker nodes, without
kube-proxy, and with ``DualStack`` enabled:

.. code-block:: shell-session

    $ cd cilium/

>     $ ./contrib/scripts/kind.sh "" 3 "" "" "none" "dual"

Reviewer 2: or just `make kind`

Author: Aye, I wanted to have `kind.sh` in the example, as it is shows that it's possible to configure a Kind cluster.

    ...
    Kind is up! Time to install cilium:
    make kind-image
    make kind-install-cilium

> Afterwards, you need to install Cilium. The preferred way is to use
> `cilium-cli install <https://github.com/cilium/cilium-cli/#install-cilium>`_,
> as it is able to automate some steps (e.g., detecting ``kube-apiserver`` endpoint
> address which otherwise needs to be specified when running w/o ``kube-proxy``, or
> set an annotation to a K8s worker node to prevent Cilium from being scheduled on it).

Reviewer: Clarity, no Latin abbreviations
```suggestion
After installing kind, the next step is to install Cilium. The preferred installation method is to use
`cilium-cli install <https://github.com/cilium/cilium-cli/#install-cilium>`_,
which can automate some steps (for example, detecting a ``kube-apiserver`` endpoint
address which otherwise needs to be specified when running without ``kube-proxy``; or
setting an annotation to a Kubernetes worker node to prevent Cilium from being scheduled on it).
```

> Assuming that Cilium was built and pushed to a local registry with:

Reviewer 2: I don't think we use the local registry anymore since https://github.com/cilium/cilium/pull/24541
